pages/icons.js

```diff
@@ -7,65 +7,87 @@ import IconCard from "@components/IconCard";
 import Page from "@components/Page";
 import PageHead from "@components/PageHead";
 import { PROJECT_NAME } from "@constants/index";
+import { useRouter } from "next/router";
+import Button from "@components/Button";
 
-export default function Icons() {
-  const [searchedIconNames, setSearchedIconNames] = useState([]);
-  const [notFound, setNotFound] = useState();
-  const [threeOrMore, setThreeOrMore] = useState();
-
-  const popularIcons = [
-    "FaGithub",
-    "FaTwitter",
-    "FaLinkedin",
-    "FaGit",
-    "FaXTwitter",
-    "FaInstagram",
-    "SiHashnode",
-    "FaLink",
-    "FaYoutube",
-    "FaGlobe",
-    "FaDev",
-    "FaDiscord",
-    "FaMedium",
-    "SiMedium",
-    "FaFacebook",
-    "FaGithubAlt",
-    "SiLinkedin",
-    "SiLeetcode",
-    "FaDollarSign",
-    "FaMastodon",
-  ];
-  const icons = {};
+const icons = {};
 
-  Object.keys(FaIcons).forEach((key) => {
-    icons[key.toLocaleLowerCase()] = key;
-  });
+Object.keys(FaIcons).forEach((key) => {
+  icons[key.toLocaleLowerCase()] = key;
+});
 
-  Object.keys(SiIcons).forEach((key) => {
-    icons[key.toLocaleLowerCase()] = key;
-  });
+Object.keys(SiIcons).forEach((key) => {
+  icons[key.toLocaleLowerCase()] = key;
+});
 
-  const searchIcons = (value) => {
-    setSearchedIconNames([]);
-    if (value.length < 3) {
-      return setThreeOrMore(false);
-    }
-    setThreeOrMore(true);
+const popularIcons = [
+  "FaGithub",
+  "FaTwitter",
+  "FaLinkedin",
+  "FaGit",
+  "FaXTwitter",
+  "FaInstagram",
+  "SiHashnode",
+  "FaLink",
+  "FaYoutube",
+  "FaGlobe",
+  "FaDev",
+  "FaDiscord",
+  "FaMedium",
+  "SiMedium",
+  "FaFacebook",
+  "FaGithubAlt",
+  "SiLinkedin",
+  "SiLeetcode",
+  "FaDollarSign",
+  "FaMastodon",
+];
 
-    const filteredIconNames = Object.keys(icons)
-      .filter((icon) => icon.includes(value.toLocaleLowerCase()))
-      .map((iconName) => icons[iconName]);     
-    if (!filteredIconNames.length) {
-      return setNotFound(value);
-    }
+export default function Icons() {
+  const [notFound, setNotFound] = useState();
 
-    if (filteredIconNames.length) {
-      setNotFound();
-    }
+  const router = useRouter();
+  const keyword = router.query.keyword || "";
+  const [threeOrMore, setThreeOrMore] = useState(
+    keyword.length >= 3 ? true : false,
+  );
+  const [searchQuery, setSearchQuery] = useState(keyword);
 
-    setSearchedIconNames(filteredIconNames);
+  const handleSubmit = (e) => {
+    e.preventDefault();
```

Comment:
```suggestion
    e.preventDefault();
        if (searchQuery === keyword) return;
```
I forgot to add one check here.
There is no need to update the URL if  the values are  same it will prevent unnecessary reruns

Reply:
Ok thank you 👍